predicate.go

```diff
@@ -0,0 +1,219 @@
+		var buf bytes.Buffer
+		body := io.TeeReader(r.Body, &buf)
+		glog.V(2).Infof("body: %s", buf.String())
+
+		var extenderArgs schedulerapi.ExtenderArgs
+		var extenderFilterResult *schedulerapi.ExtenderFilterResult
+
+		if err := json.NewDecoder(body).Decode(&extenderArgs); err != nil {
+			extenderFilterResult = &schedulerapi.ExtenderFilterResult{
+				Nodes:       nil,
+				FailedNodes: nil,
+				Error:       err.Error(),
+			}
+		} else {
+			extenedResourceScheduler := &ExtendedResourceScheduler{
+				Clientset: clientset,
+			}
+			extenderFilterResult = filter(extenderArgs, extenedResourceScheduler)
+		}
+
+		w.Header().Set("Content-Type", "application/json")
+		if resultBody, err := json.Marshal(extenderFilterResult); err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			w.Write([]byte(err.Error()))
+		} else {
+			w.WriteHeader(http.StatusOK)
+			w.Write(resultBody)
+		}
+	}
+}
+
+// Bind delegates the action of binding a pod to a node to the extender.
+func Bind(clientset *kubernetes.Clientset) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		var buf bytes.Buffer
+		body := io.TeeReader(r.Body, &buf)
+		var extenderBindingArgs schedulerapi.ExtenderBindingArgs
+
+		if err := json.NewDecoder(body).Decode(&extenderBindingArgs); err != nil {
+			glog.Errorf("json decode error: %v", err)
+			w.Header().Set("Content-Type", "application/json")
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte(err.Error()))
+		}
+		b := &v1.Binding{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: extenderBindingArgs.PodNamespace,
+				Name:      extenderBindingArgs.PodName,
+				UID:       extenderBindingArgs.PodUID,
+			},
+			Target: v1.ObjectReference{
+				Kind: "Node",
+				Name: extenderBindingArgs.Node,
+			},
+		}
+		err := clientset.CoreV1().Pods(b.Namespace).Bind(b)
+		if err != nil {
+			glog.Errorf("bind failed: %v", err)
+			w.Header().Set("Content-Type", "application/json")
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte(err.Error()))
+		}
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+	}
+}
+
+func filter(extenderArgs schedulerapi.ExtenderArgs, extenedResourceScheduler *ExtendedResourceScheduler) *schedulerapi.ExtenderFilterResult {
+	pod := extenderArgs.Pod
+	nodes := extenderArgs.Nodes.Items
+
+	canSchedule := make([]v1.Node, 0, len(extenderArgs.Nodes.Items))
+	canNotSchedule := make(map[string]string)
+	// default all node scheduling failed
+	defaultNotSchedule := defaultFailedNodes(nodes)
+
+	result := &schedulerapi.ExtenderFilterResult{
+		Nodes: &v1.NodeList{
+			Items: canSchedule,
+		},
+		FailedNodes: defaultNotSchedule,
+		Error:       "",
+	}
+
+	ercs, err := findExtendedResourceClaims(pod, extenedResourceScheduler)
+	if err != nil {
+		glog.Errorf("find extendedresourceclaims failed: %v", err)
+		result.Error = err.Error()
+		return result
+	}
+	// Find ExtendedResourceClaim by extendedresourceclaim's name
```

Comment:
what does this comment mean ?

Reply:
sorry, I forget to delete this comment, I will delete soon